qiskit_experiments/library/characterization/analysis/tphi_analysis.py

```diff
@@ -34,23 +53,24 @@ def _run_analysis(
         _run_analysis for the two sub-experiments.
         Based on the results, it computes the result for :math:`T_phi`.
         """
-        super()._run_analysis(experiment_data)
+        # Run composite analysis and extract T1 and T2star results
+        analysis_results, figures = super()._run_analysis(experiment_data)
+        t1_result = next(filter(lambda res: res.name == "T1", analysis_results))
```

Comment:
This makes me think current framework is bit problematic. If we write a composite analysis wrapper class that offers the capability to choose combine policy, we need to write
```python
if self.combine_results:
    # extract from experiment data
else:
    # filter analysis result
```
I think we should update the logic of `_combine_component_results` so that it can take both `component_experiment_data` and `experiment_data` then it copies corresponding result and figure entries from component to self experiment data.

Reply:
Yeah its a bit tricky if you want to make a custom subclass that isn't always combined. What you suggest would also be a bit problematic since it bypasses the functionality of BaseAnalysis which does some post-processing of the returns from `_run_analysis` to set the correct experiment id and components.

I was thinking of adding some extra public methods to CompositeAnalysis to extract/generate the component experiment data that a user/developer could call directly. It will lose a small amount of efficiency, but maybe thats ok.

Reply:
Basically, this problem will be addressed if we define `_extra_entry` method in base analysis, as curve analysis does. This method should be called after `_run_analysis` is completed rather than within the `_run_analysis` as curve analysis does.